File: Backend/spring-boot-library/src/main/java/com/reactlibraryproject/springbootlibrary/Entity/PurchaseHistory.java
package com.reactlibraryproject.springbootlibrary.Entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "purchase_history")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PurchaseHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    private String userEmail;

    private String title;

    private String author;

    private String category;

    private String img;

    private String publisher;

    private int amount;

    private int price;

    private String paymentKey;

    private String purchaseDate;

    private String orderId;

    private String status;

    private Long cartItemId;
}

File: Frontend/library-app/src/models/PurchaseHistory.ts
class PurchaseHistory {
  paymentKey: string;
  orderName: string;
  totalPrice: number;
  purchaseDate: string;
  orderId: string;
  constructor(
    paymentKey: string,
    oderName: string,
    totalPrice: number,
    purchaseDate: string,
    orderId: string
  ) {
    this.paymentKey = paymentKey;
    this.orderName = oderName;
    this.totalPrice = totalPrice;
    this.purchaseDate = purchaseDate;
    this.orderId = orderId;
  }
}
export default PurchaseHistory;
